### DISCUSSION TITLE
@CustomScalar and GraphQL schema

### DISCUSSION THREAD

**Discussion by ehuisman** [smallrye/smallrye-graphql#2174] [Q&A] [answered]
I've implemented a custom scalar as per the [documentation](https://smallrye.io/smallrye-graphql/latest/custom-scalar/), but I'm struggling to get it included in the generated GraphQL schema. I've tried setting the `smallrye.graphql.schema.includeScalars` (well, actually `quarkus.smallrye-graphql.schema-include-scalars`, since I'm using Quarkus) configuration property, but based on my research/debugging, it only affects the printing of the schema, not the actual mapping from `io.smallrye.graphql.schema.model.Schema` to `graphql.schema.GraphQLSchema` (of `graphql-java`).

It seems that `io.smallrye.graphql.bootstrap.Bootstrap` does register custom scalar types to `GraphQLScalarTypes`, but they are never passed to `GraqhQLSchema.Builder` as `additionalTypes`, thus they are only used as type names, but not included in the schema as `scalar`s. This severely limits the ability to document how a custom scalar is supposed to be used.

Is there something I'm missing, or should I file an issue about this?

**Comment by jmartisk** [comment]
That is odd. Would you be willing to create a simple reproducer project to make the investigation easier? We'd try to get to it next week or so

**Reply by ehuisman** [reply, marked as answer]
I tried creating a simple reproducer, couldn't reproduce, and after trying to figure out why, now the original one also works. No idea why it didn't earlier, but the problem is likely to be between keyboard and chair.
